kernel/src/thread/thread_table.rs

```diff
@@ -1,12 +1,9 @@
 // SPDX-License-Identifier: MPL-2.0
 
-use keyable_arc::KeyableArc;
-
 use super::{Thread, Tid};
 use crate::{prelude::*, process::posix_thread::PosixThreadExt};
 
 static POSIX_THREAD_TABLE: SpinLock<BTreeMap<Tid, Arc<Thread>>> = SpinLock::new(BTreeMap::new());
-static KERNEL_THREAD_TABLE: SpinLock<BTreeSet<KeyableArc<Thread>>> = SpinLock::new(BTreeSet::new());
```

Comment:
Now that `thread_table` only manages POSIX threads, I think this file should be moved to `process/posix_thread/`. And because the module belongs to `posix_thread`, the API can be simply named `get_thread` and `add_thread` (no posix).